trtools/utils/tr_harmonizer.py

```diff
@@ -1141,10 +1147,19 @@ def GetDosages(self,
 
             # Check AP field. allow wiggle room for rounding
             if np.any(np.sum(ap1, axis=1) > 1.1) or np.any(np.sum(ap2, axis=1) > 1.1):
-                raise ValueError("AP1 or AP2 field summing to more than 1 detected")
+                error_msg = "{}:{} AP1 or AP2 field summing to more than 1 detected".format(self.chrom, self.pos)
+                if strict:
+                    raise ValueError(error_msg)
+                else:
+                    common.WARNING(error_msg)
+                    return np.array([np.nan]*self.GetNumSamples())
```

Comment:
At the moment, this will return NaN for all samples even if only one had an improper AP value. Would it be better to only return NaN for the few bad samples, instead?
```suggestion
            failing = (np.sum(ap1, axis=1) > 1.1) | (np.sum(ap2, axis=1) > 1.1)
            if np.any(failing):
                error_msg = "{}:{} AP1 or AP2 field summing to more than 1 detected".format(self.chrom, self.pos)
                if strict:
                    raise ValueError(error_msg)
                else:
                    common.WARNING(error_msg)
                    ap1[failing,:] = np.nan
                    ap2[failing,:] = np.nan
```